connectors/es/sink.py

```diff
@@ -170,6 +179,48 @@ async def _bulk_api_call():
 
         return res
 
+    async def _log_bulk_operations(self, res):
+        for item in res.get("items", []):
+            if "index" in item:
+                action_item = "index"
+            elif "delete" in item:
+                action_item = "delete"
+            elif "create" in item:
+                action_item = "create"
+            elif "update" in item:
+                action_item = "update"
+            else:
+                # Should only happen, if the _bulk API changes
+                # Unlikely, but as this functionality could be used for audits we want to detect changes fast
+                self._logger.error(
+                    f"Unknown action item returned from _bulk API for item {item}"
+                )
+                return
+
+            doc_id = item[action_item].get("_id")
+            if doc_id is None:
+                # Should only happen, if the _bulk API changes
+                # Unlikely, but as this functionality could be used for audits we want to detect changes fast
+                self._logger.error(f"Could not retrieve '_id' for document {item}")
+                return
```

Comment:
edge case worth validating - a pipeline removes `_id` and then a drop processor drops the doc, so it isn't assigned a random ID by Elasticsearch. IDK what the response for that record would look like.